### PR TITLE
refactor(uuid): [noticket]: Replace uuid package with generateId util

### PR DESCRIPTION
**Pull request by diogomateus**
### What this PR does
Replaces uuid package with a custom and simpler generateId util.

### Why is this needed?
uuid package can't be tree-shaked and was adding a giant additional size to repos where it was being imported.
Usually, using random for random ids is not the most efficient way but since this ids are not persistent and used for runtime only (generate label/input ids, temporary ids for error entries) this downgrade from uuid might make sense.

Named the util generateId so it doesn't suggest it is a fully and standardised uuid and replaces the function.
The id generated is based on a random number that is then parsed into an hex string for each of the characters.

### Checklist:

- [ ] I reviewed my own code
- [ ] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [ ] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [ ] I have updated the task(s) status on Linear
- [ ] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by diogomateus** [comment]
@everyone I'm very open to other proposals regarding this refactoring.

File: src/lib/util/generateId/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import generateId from '.';
+
+describe('generateId', () => {
+  it('should return two different ids', () => {
+    const firstId = generateId();
+    const secondId = generateId();
+
+    expect(firstId).not.toEqual(secondId);
+  });
```
Comment (talo242) [inline src/lib/util/generateId/index.test.ts:9]: Not sure if it'd be worth it to write regression tests for the format `xxxx-xxxx-xxx-xxxx` and/or length of the string? (only if it's relevant)

Comment (diogomateus) [reply src/lib/util/generateId/index.test.ts:9]: You regression as a way to make sure it's compatible with previous usages?

**Comment by kimkwanka** [comment]
Instead of cooking up our own solution here which, in theory, could still produce identical ids, would maybe using https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID be an alternative here? It looks like browser support is solid

File: src/lib/util/generateId/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+function generateId(): string {  
+  if (!window?.crypto?.randomUUID) {
```
Comment (diogomateus) [inline src/lib/util/generateId/index.ts:2]: Thanks for that suggestion, @kimkwanka!

Added a fallback for the edge cases although support is over 90% of browsers, wdyt?

Comment (kimkwanka) [reply src/lib/util/generateId/index.ts:2]: Sounds great 👍

**Review by kimkwanka — APPROVED** [review]

**Comment by diogomateus** [comment]
Seems like `randomUUID` runs into a lot of issues and is only available in [secure contexts](https://github.com/WICG/uuid/issues/23) meaning we might run into some issues using it. On top of that it's only available on node v19 and we're running on v16 which means it triggers errors when building storybook.

Updated it to use this function by broofa that is one of the contributors to the uuid library and, I believe, part of the team that implemented randomUUID.

Still defaulting to Math random for places where we might run into an issue. Testing should be one those cases.

**Comment by kimkwanka** [comment]
> Seems like `randomUUID` runs into a lot of issues and is only available in [secure contexts](https://github.com/WICG/uuid/issues/23) meaning we might run into some issues using it. On top of that it's only available on node v19 and we're running on v16 which means it triggers errors when building storybook.
> 
> Updated it to [use this function by broofa ](https://stackoverflow.com/a/2117523)that is one of the contributors to the uuid library and, I believe, part of the team that implemented randomUUID.
> 
> Still defaulting to Math random for places where we might run into an issue. Testing should be one those cases.

That's a shame, but I guess that's all that can be done at this point. Thanks 👍

**Review by kimkwanka — APPROVED** [review]

**Comment by diogomateus** [comment]
> > Seems like `randomUUID` runs into a lot of issues and is only available in [secure contexts](https://github.com/WICG/uuid/issues/23) meaning we might run into some issues using it. On top of that it's only available on node v19 and we're running on v16 which means it triggers errors when building storybook.
> > Updated it to [use this function by broofa ](https://stackoverflow.com/a/2117523)that is one of the contributors to the uuid library and, I believe, part of the team that implemented randomUUID.
> > Still defaulting to Math random for places where we might run into an issue. Testing should be one those cases.
> 
> That's a shame, but I guess that's all that can be done at this point. Thanks 👍

100%, pretty annoying that we have to go through such hassle just to create an ID when there's even a function already for it 😅

**Comment by broofa** [comment]
> uuid package can't be tree-shaked

Can you elaborate on this?  The `uuid` module was refactored a while back precisely to facillitate tree-shaking, so this comment is  a little surprising.

**Comment by kimkwanka** [comment]
> > uuid package can't be tree-shaked
> 
> Can you elaborate on this? The `uuid` module was refactored a while back precisely to facillitate tree-shaking, so this comment is a little surprising.

@diogomateus

**Comment by diogomateus** [comment]
> > uuid package can't be tree-shaked
> 
> Can you elaborate on this? The `uuid` module was refactored a while back precisely to facillitate tree-shaking, so this comment is a little surprising.

After digging up a bit, I think this might be related to the fact that we were using the v0.8.3 instead of the most recent one where the minified UMD build was removed so there shouldn't be an issue with it. I'll take a look at replicating it again later.

**Pull request merged**